x/mongo/driver/batch_cursor.go

```diff
@@ -241,7 +241,7 @@ func (bc *BatchCursor) getMore(ctx context.Context) {
 
 	// Required for legacy operations which don't support limit.
 	numToReturn := bc.batchSize
-	if bc.limit != 0 && bc.numReturned+bc.batchSize > bc.limit {
+	if bc.limit != 0 && bc.numReturned+bc.batchSize >= bc.limit {
```

Comment:
Based on my reading of https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst#combining-limit-and-batch-size-for-the-wire-protocol I believe it is ok to use `batchSize=0` when a limit is set. So I think this might be ok? I'm still not entirely sure where the limit is enforced after, when `batchSize=0` and more than the limit documents are returned. I think tests which query a collection containing more than 101 documents may verify that behavior.

Reply:
I'm not sure either, but the tests now query a collection containing more than 101 documents and pass...

Reply:
Great, the new tests cover the case of a getMore being run.